python/cudf/cudf/pylibcudf_tests/test_string_replace.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+import cudf._lib.pylibcudf as plc
+
+
+@pytest.fixture(scope="module")
+def pa_data_col():
+    return pa.array(["a", "c", "A", "aa", None, "aaaaaaaaa", "AAAA", "ÁÁÁÁ"])
+
+
+@pytest.fixture(scope="module")
+def plc_data_col(pa_data_col):
+    return plc.interop.from_arrow(pa_data_col)
+
+
+@pytest.fixture(scope="module", params=["a", "c", "A", "Á", "aa", "ÁÁÁ"])
+def scalar_repl_target(request):
+    pa_target = pa.scalar(request.param, type=pa.string())
+    return (request.param, plc.interop.from_arrow(pa_target))
```

Comment:
Interesting approach. This accomplishes the same thing in spirit that the other tests have done with a plc fixture that's constructed from the arrow fixture, just in a slightly different way. The more I stare at it, I'm starting to prefer this approach a little since it ties the two objects very tightly together in a way that reflects that they'll probably always be used together in tests. I suppose it's largely a matter of taste though. @mroeschke @brandon-b-miller WDYT? I would generally prefer to be consistent across tests; it's not a huge deal, but given that our test suite is still fairly small I'd be fine with a quick PR to standardize on this approach if we like it better (or change this to the separate fixture approach if we prefer that).

Reply:
This is what we do in pandas.

The advantage of this approach is being able to pair the values together.
(which is good since you'll never have cases where the fixtures get out of sync)

The disadvantage is that you have to add a line unpacking the tuple in every test that uses the fixture.

I'm happy with either, just did it this way since it was more familiar for me.

Reply:
I do think I prefer this approach. After this PR merges, could you make a follow-up that standardizes this in other tests? Basically just removing pairs of (pyarrow,pylibcudf) fixtures in favor of a single fixture returning the pair?

Reply:
Will do.